### PR TITLE
Integrate message validation hook to multi onramp

### PR DESCRIPTION
**Pull request by RayXpub**
## Motivation

Follow up of https://github.com/smartcontractkit/ccip/pull/916. 

## Solution

- Replaces the `EVM2EVMMultiOnRamp` `AggregateRateLimiter` inheritance by a call to an `IMessageValidator` hook contract to validate messages.
- Implements the `MultiAggregateRateLimiter.onOutgoingMessage()` function.

### CODE REVIEW

File: contracts/src/v0.8/ccip/test/helpers/MessageInterceptorHelper.sol [hunk 1]
```diff
@@ -16,15 +16,17 @@ contract MessageInterceptorHelper is IMessageInterceptor {
...
       revert IncomingMessageValidationError(bytes("Invalid message"));
     }
   }
 
   /// @inheritdoc IMessageInterceptor
-  function onOutgoingMessage(Client.EVM2AnyMessage memory, uint64) external pure {
-    // TODO: to be implemented
+  function onOutgoingMessage(uint64, Client.EVM2AnyMessage calldata message) external view {
+    if (s_invalidMessageIds[keccak256(abi.encode(message))]) {
+      revert IncomingMessageValidationError(bytes("Invalid message"));
```
Comment (elatoskinas) [inline contracts/src/v0.8/ccip/test/helpers/MessageInterceptorHelper.sol]: nit: we should rename the error to something like `MessageValidationError`

File: contracts/src/v0.8/ccip/onRamp/EVM2EVMMultiOnRamp.sol [hunk 2]
```diff
@@ -347,15 +352,13 @@ contract EVM2EVMMultiOnRamp is IEVM2AnyMultiOnRamp, AggregateRateLimiter, ITypeA
...
     // Only check token value if there are tokens
     if (numberOfTokens > 0) {
-      uint256 value;
-      for (uint256 i = 0; i < numberOfTokens; ++i) {
-        if (message.tokenAmounts[i].amount == 0) revert CannotSendZeroTokens();
-        if (s_tokenTransferFeeConfig[destChainSelector][message.tokenAmounts[i].token].aggregateRateLimitEnabled) {
-          value += _getTokenValue(message.tokenAmounts[i], IPriceRegistry(s_dynamicConfig.priceRegistry));
+      address messageValidator = s_dynamicConfig.messageValidator;
+      if (messageValidator != address(0)) {
+        try IMessageInterceptor(s_dynamicConfig.messageValidator).onOutgoingMessage(destChainSelector, message) {}
```
Comment (elatoskinas) [inline contracts/src/v0.8/ccip/onRamp/EVM2EVMMultiOnRamp.sol]: suggestion: should use the cached `messageValidator` here

File: contracts/src/v0.8/ccip/test/rateLimiter/MultiAggregateRateLimiter.t.sol [hunk 3]
```diff
@@ -1,17 +1,15 @@
...
 import {MultiAggregateRateLimiter} from "../../MultiAggregateRateLimiter.sol";
 import {Client} from "../../libraries/Client.sol";
 import {Internal} from "../../libraries/Internal.sol";
 import {RateLimiter} from "../../libraries/RateLimiter.sol";
 import {BaseTest} from "../BaseTest.t.sol";
 import {MultiAggregateRateLimiterHelper} from "../helpers/MultiAggregateRateLimiterHelper.sol";
 import {PriceRegistrySetup} from "../priceRegistry/PriceRegistry.t.sol";
-
 import {stdError} from "forge-std/Test.sol";
+import {Vm} from "forge-std/Vm.sol";
```
Comment (elatoskinas) [inline contracts/src/v0.8/ccip/test/rateLimiter/MultiAggregateRateLimiter.t.sol:14]: suggestion: we should implement this TODO in the file to make sure that the incoming and outgoing bucket states are separate:

```
  // TODO: add a test case similar to this one to verify that onIncomingMessage / onOutgoingMessage rate limits
  // are applied to 2 different buckets
```

File: contracts/src/v0.8/ccip/onRamp/EVM2EVMMultiOnRamp.sol [hunk 4]
```diff
@@ -835,6 +838,20 @@ contract EVM2EVMMultiOnRamp is IEVM2AnyMultiOnRamp, AggregateRateLimiter, ITypeA
   // ================================================================
   // │                           Access                             │
   // ================================================================
+  /// @notice Gets the admin address.
+  /// @return the admin address.
+  function getAdmin() external view returns (address) {
```
Comment (elatoskinas) [inline contracts/src/v0.8/ccip/onRamp/EVM2EVMMultiOnRamp.sol]: Let's create a ticket to revisit if we need the admin on the OnRamp, there was a thread on potentially removing this but we did not close on it

File: contracts/.changeset/sour-toys-drive.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+'@chainlink/contracts-ccip': minor
```
Comment (RensR) [inline contracts/.changeset/sour-toys-drive.md]: nit: remove

**Review by connorwstein — APPROVED** [review]

File: contracts/.changeset/sour-toys-drive.md (hunk 5 above)
Comment (RayXpub) [reply contracts/.changeset/sour-toys-drive.md]: Removed

File: contracts/src/v0.8/ccip/test/helpers/MessageInterceptorHelper.sol (hunk 1 above)
Comment (RayXpub) [reply contracts/src/v0.8/ccip/test/helpers/MessageInterceptorHelper.sol]: Done, reused the error from the `IMessageInterceptor`.

File: contracts/src/v0.8/ccip/onRamp/EVM2EVMMultiOnRamp.sol (hunk 2 above)
Comment (RayXpub) [reply contracts/src/v0.8/ccip/onRamp/EVM2EVMMultiOnRamp.sol]: Good catch, updated.

File: contracts/src/v0.8/ccip/test/rateLimiter/MultiAggregateRateLimiter.t.sol (hunk 3 above)
Comment (RayXpub) [reply contracts/src/v0.8/ccip/test/rateLimiter/MultiAggregateRateLimiter.t.sol:14]: Done https://github.com/smartcontractkit/ccip/blob/0ff51fc2620308340147680abb18d37ce2dc4c20/contracts/src/v0.8/ccip/test/rateLimiter/MultiAggregateRateLimiter.t.sol#L1276

File: contracts/src/v0.8/ccip/onRamp/EVM2EVMMultiOnRamp.sol (hunk 4 above)
Comment (RayXpub) [reply contracts/src/v0.8/ccip/onRamp/EVM2EVMMultiOnRamp.sol]: Created

**Review by elatoskinas — APPROVED** [review]

**Pull request merged**